CRM/MembersOnlyEvent/Service/MembersOnlyEventAccess.php

```diff
@@ -62,18 +62,25 @@ public function __construct($eventID) {
   public function userHasEventAccess() {
     if (
       empty($this->membersOnlyEvent) ||
-      CRM_Core_Permission::check('members only event registration')
+      ($this->contactID && CRM_Core_Permission::check('members only event registration'))
     ) {
       // Any user (including anonymous) with 'members only event registration' permission
       // can access any members-only event.
       return TRUE;
     }
 
-    if (empty($this->membersOnlyEvent->is_groups_only)) {
+    $event_access_type = (int) $this->membersOnlyEvent->event_access_type;
```

Comment:
I wonder this again, why we need to cast this to `int `, since the value of `event_access_type` should already be int?

Reply:
This was necessary to make some tests pass.